### PR TITLE
Introducing Status Pollers for Bosh Based Restore

### PR DESCRIPTION
**Pull request by jayeshjk**

### CODE REVIEW

File: operators/bosh-restore-operator/BoshRestoreService.js [hunk 1]
```diff
@@ -90,7 +90,7 @@ class BoshRestoreService extends BaseDirectorService {
         resourceId: opts.restore_guid,
         options: optionsData,
         status: {
-          'state': `${CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS}_BOSH_STOP`,
+          'state': `${CONST.APISERVER.RESOURCE_STATE.TRIGGER}_BOSH_STOP`,
           'response': data
         }
       });
```
Comment (amitmalav) [inline operators/bosh-restore-operator/BoshRestoreService.js]: Is this catch really necessary? There is a catch already present in [DefaultBoshRestoreOperator](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/648/files#diff-f12ac58b1cf691827c26f10a44281436L37)

File: operators/bosh-restore-operator/BoshRestoreStatusPoller.js [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+      resourceGroup: CONST.APISERVER.RESOURCE_GROUPS.RESTORE,
+      resourceType: CONST.APISERVER.RESOURCE_TYPES.DEFAULT_BOSH_RESTORE,
+      validStateList: [`${CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS}_BOSH_STOP`,
+        `${CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS}_ATTACH_DISK`,
+        `${CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS}_BASEBACKUP_ERRAND`,
+        `${CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS}_PITR_ERRAND`,
+        `${CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS}_BOSH_START`,
+        `${CONST.APISERVER.RESOURCE_STATE.IN_PROGRESS}_POST_BOSH_START`],
+      validEventList: [CONST.API_SERVER.WATCH_EVENT.ADDED, CONST.API_SERVER.WATCH_EVENT.MODIFIED],
+      pollInterval: 10000 // TODO: Take it from constant
```
Comment (amitmalav) [inline operators/bosh-restore-operator/BoshRestoreStatusPoller.js]: TODO :p

File: operators/bosh-restore-operator/BoshRestoreService.js (hunk 1 above)
Comment (jayeshjk) [reply operators/bosh-restore-operator/BoshRestoreService.js]: yep, this `try...catch` can be removed.

File: operators/bosh-restore-operator/BoshRestoreStatusPoller.js (hunk 2 above)
Comment (jayeshjk) [reply operators/bosh-restore-operator/BoshRestoreStatusPoller.js]: Done.

File: operators/bosh-restore-operator/BoshRestoreService.js (hunk 1 above)
Comment (jayeshjk) [reply operators/bosh-restore-operator/BoshRestoreService.js]: Done

File: common/constants.js [hunk 3]
```diff
@@ -376,7 +376,9 @@ module.exports = Object.freeze({
       ABORTED: 'aborted',
       UPDATE: 'update',
       LOCKED: 'locked',
-      UNLOCKED: 'unlocked'
+      UNLOCKED: 'unlocked',
```
[the remark this answers is not in the document]
Comment (jayeshjk) [reply common/constants.js]: Done

**Pull request merged**